Update use-tidb-to-read-tiflash.md (#15866)
pingcap/docs: tiflash/use-tidb-to-read-tiflash.md

> **Note:**
>
> - Before v4.0.3, the behavior of reading from TiFlash replica in a non-read-only SQL statement (for example, `INSERT INTO ... SELECT`, `SELECT ... FOR UPDATE`, `UPDATE ...`, `DELETE ...`) is undefined.

> > - In versions between v4.0.3 and v6.3.0, internally TiDB ignores the TiFlash replica for a non-read-only SQL statement to guarantee the data correctness. That is, for [smart selection](#smart-selection), TiDB automatically selects the non-TiFlash replica; for [engine isolation](#engine-isolation) that specifies TiFlash replica **only**, TiDB reports an error; and for [manual hint](#manual-hint), TiDB ignores the hint.

Author: ```suggestion
> - In versions between v4.0.3 and v6.2.0, internally TiDB ignores the TiFlash replica for a non-read-only SQL statement to guarantee the data correctness. That is, for [smart selection](#smart-selection), TiDB automatically selects the non-TiFlash replica; for [engine isolation](#engine-isolation) that specifies TiFlash replica **only**, TiDB reports an error; and for [manual hint](#manual-hint), TiDB ignores the hint.
```

> > - In versions between v4.0.3 and v6.2.0, internally TiDB ignores the TiFlash replica for a non-read-only SQL statement to guarantee the data correctness. That is, for [smart selection](#smart-selection), TiDB automatically selects the non-TiFlash replica; for [engine isolation](#engine-isolation) that specifies TiFlash replica **only**, TiDB reports an error; and for [manual hint](#manual-hint), TiDB ignores the hint.

Reviewer: ```suggestion
> - For versions from v4.0.3 to v6.2.0, internally TiDB ignores the TiFlash replica for a non-read-only SQL statement to guarantee the data correctness. That is, for [smart selection](#smart-selection), TiDB automatically selects the non-TiFlash replica; for [engine isolation](#engine-isolation) that specifies TiFlash replica **only**, TiDB reports an error; and for [manual hint](#manual-hint), TiDB ignores the hint.
```

> > - In versions between v6.4.0 and v7.0.0, if the TiFlash replica is enabled, you can use the variable [tidb_enable_tiflash_read_for_write_stmt](/system-variables.md#tidb_enable_tiflash_read_for_write_stmt-new-in-v630) to decide whether to use the TiFlash replica for a non-read-only SQL statement.

Author: ```suggestion
> - In versions between v6.3.0 and v7.0.0, if the TiFlash replica is enabled, you can use the variable [tidb_enable_tiflash_read_for_write_stmt](/system-variables.md#tidb_enable_tiflash_read_for_write_stmt-new-in-v630) to decide whether to use the TiFlash replica for a non-read-only SQL statement.
```

> > - In v7.1.0 and later versions, if the TiFlash replica is enabled and the [SQL Mode](/sql-mode.md) of the current session is not STRICT Mode(the variable `sql_mode` does not include `STRICT_TRANS_TABLES` or `STRICT_ALL_TABLES`), TiDB will decide automatically whether to use the TiFlash replica based on cost estimation for a non-read-only SQL statement.

Reviewer: ```suggestion
> - Starting from v7.1.0, if the TiFlash replica is enabled and the [SQL Mode](/sql-mode.md) of the current session is not strict (which means the `sql_mode` value does not contain `STRICT_TRANS_TABLES` or `STRICT_ALL_TABLES`), TiDB automatically decides whether to use the TiFlash replica for a non-read-only SQL statement based on cost estimation.
```
